### DISCUSSION TITLE
lf directory history via fzf

### DISCUSSION THREAD

**Discussion by techeddie** [gokcehan/lf#1429] [Q&A] [answered]
Hello, 
can someone help me to implement a lf directory history? 
I'm using fzf_jump to jump to a sub folder. Everytime I switch into a directory this should be stored and accessed via fzf.

```bash
  cmd fzf_jump ${{
      # res="$(find . -maxdepth 1 | fzf --reverse --header='Jump to location' | sed 's/\\/\\\\/g;s/"/\\"/g')"
      res="$(fd | fzf --reverse --header='Jump to location' | sed 's/\\/\\\\/g;s/"/\\"/g')"
      if [ -d "$res" ] ; then
          cmd="cd"
      elif [ -f "$res" ] ; then
          cmd="select"
      else
          exit 0
      fi
      lf -remote "send $id $cmd \"$res\""
  }}
```

thank you

**Comment by DusanLesan** [comment, marked as answer]
Something like this maybe:
```
cmd fzf_jump_history ${{
	res="$(cat /tmp/lf_dir_history | fzf --reverse --header='Jump to location' | sed 's/\\/\\\\/g;s/"/\\"/g')"
	if [ -d "$res" ] ; then
		cmd="cd"
	elif [ -f "$res" ] ; then
		cmd="select"
	else
		exit 0
	fi

	lf -remote "send $id $cmd \"$res\""
}}

cmd on-cd &{{
	list="$(tail -n 20 /tmp/lf_dir_history | sed "\|^$PWD$|d")"
	echo "$list"$'\n'"$PWD" > /tmp/lf_dir_history
}}

map b :fzf_jump_history
```
On every change it adds one last entry to a file in tmp, removes duplicates and removes older lines. On history command it displays content of tmp file

**Reply by techeddie** [reply]
Awesome! Excatly what I was looking for.
Thank you

**Reply by CAESIUS-TIM** [reply]
Is there an extra `$` in cmd `on-cd`?

<img width="200" alt="image" src="https://github.com/gokcehan/lf/assets/60285058/de351319-9511-463d-b621-4625b32e1b42">

```diff
cmd on-cd &{{
	list="$(tail -n 20 /tmp/lf_dir_history | sed "\|^$PWD$|d")"
-	echo "$list"$'\n'"$PWD" > /tmp/lf_dir_history
+	echo "$list"'\n'"$PWD" > /tmp/lf_dir_history
}}

```

**Reply by CAESIUS-TIM** [reply]
This change make a better look in "which-key".
```diff
-map b :fzf_jump_history
+map b fzf_jump_history
```
before:
![image](https://github.com/gokcehan/lf/assets/60285058/b84c8ab5-6d24-492d-88f3-02159a5eb6f7)
after:
![image](https://github.com/gokcehan/lf/assets/60285058/96f204e9-4f5a-4a50-aae2-bcd38938f5dd)

By the way, fzf_select: https://github.com/gokcehan/lf/issues/705

**Reply by CAESIUS-TIM** [reply]
And is `Ctrl+c` the only way to exit both fzf cmd above and stay in lf without select?

**Reply by DusanLesan** [reply]
This could be shell specific. I think I've needed to use `$'\n'` to echo new lines without adding any parameters to it
You can bind actions to custom fzf keys. I like using esc for such cases

**Comment by techeddie** [comment]
resolved
